lib/value/src/value.rs

```diff
@@ -769,6 +770,32 @@ impl Value {
         }
     }
 
+    /// Returns a reference to a field value specified by a path iter.
+    pub fn get_by_path_v2<'a>(&self, path: impl Path<'a>) -> Option<&Self> {
+        let mut value = self;
+        let mut path_iter = path.segment_iter();
+        loop {
+            match (path_iter.next(), value) {
+                (None, _) => return Some(value),
+                (Some(BorrowedSegment::Field(key)), Value::Object(map)) => match map.get(key) {
+                    None => return None,
+                    Some(nested_value) => {
+                        value = nested_value;
+                    }
+                },
+                (Some(BorrowedSegment::Index(index)), Value::Array(array)) => {
+                    match array.get(index as usize) {
+                        None => return None,
+                        Some(nested_value) => {
+                            value = nested_value;
+                        }
+                    }
+                }
+                _ => return None,
```

Comment:
Since this method allows both parsed and unparsed paths, it does look like this will hide any parsing errors right now. I worry that might lead to hard-to-find bugs?

Reply:
It is definitely an issue, but this works exactly the same as it does today. It's one of the planned future improvements listed here: https://github.com/vectordotdev/vector/issues/7070

Reply:
Ah sorry, I had seen you link to that issue, but didn't see a todo list in the issue description. I just noticed you added a comment with that list there 👍